src/game.cpp

```diff
@@ -2831,7 +2831,10 @@ void Game::toggleMinimap(bool shift_pressed)
 	MinimapMode mode = MINIMAP_MODE_OFF;
 	if (hud_flags & HUD_FLAG_MINIMAP_VISIBLE) {
 		mode = mapper->getMinimapMode();
-		mode = (MinimapMode)((int)mode + 1);
+		mode = (!(hud_flags & HUD_FLAG_MINIMAP_RADAR) &&
```

Comment:
bool isRadarAllowed = hud_flags & HUD_FLAG_MINIMAP_RADAR == 0;

it's more accurate and recommended by checkstyle tools